### PR TITLE
Added an example for advanced usage of WOW

### PR DESCRIPTION
**Pull request by frederic-auchere**
This a an example of some more advanced usage of the WOW algorithm. It demonstrates regular vs. edge-aware flavors, denoising, and merging with the input data.

### CODE REVIEW

File: examples/advanced_wow.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+###########################################################################
+# Finally, we merge the denoised edge-aware enhanced image with the gamma-stretched input, with weight h.
+
+gamma = 4
+wow_map_bilateral_denoised_merged = enhance.wow(input_map,
+                                                bilateral=1,
+                                                denoise_coefficients=denoise_coefficients,
+                                                gamma=gamma,
+                                                h=0.99)
+
```
Comment (nabobalis) [inline examples/advanced_wow.py:71]:
```suggestion
# Finally, we merge the denoised edge-aware enhanced image with the
# gamma-stretched input, with weight "h".

gamma = 4
wow_map_bilateral_denoised_merged = enhance.wow(input_map, bilateral=1, denoise_coefficients=denoise_coefficients, gamma=gamma, h=0.99)

```

File: examples/advanced_wow.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+###########################################################################
+# Now we will plot the final result and compare that to the original image.
```
Comment (nabobalis) [inline examples/advanced_wow.py:73]:
```suggestion
# Finally, we will plot the full set of outputs created and
# compare that to the original image.
```

File: examples/advanced_wow.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+fig = plt.figure(figsize=(8, 12))
+
```
Comment (nabobalis) [inline examples/advanced_wow.py:76]:
```suggestion
```

File: examples/advanced_wow.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+variations = {
+    'Input | gamma = {gamma} stretch': {'map': input_map, 'stretch': PowerStretch(1 / gamma)},
+    'WOW | linear stretch': {'map': wow_map, 'stretch': LinearStretch()},
+    'denoised WOW': {'map': wow_map_denoised, 'stretch': LinearStretch()},
+    'Edge-aware WOW': {'map': wow_map_bilateral, 'stretch': LinearStretch()},
+    'Edge-aware & denoised WOW': {'map': wow_map_bilateral_denoised, 'stretch': LinearStretch()},
+    'Merged with input': {'map': wow_map_bilateral_denoised_merged, 'stretch': LinearStretch()}
+}
+
```
Comment (nabobalis) [inline examples/advanced_wow.py:85]:
```suggestion
```

File: examples/advanced_wow.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+interval = AsymmetricPercentileInterval(1, 99.9)
+
```
Comment (nabobalis) [inline examples/advanced_wow.py:87]:
```suggestion
```

**Review by nabobalis — APPROVED** [review]
Some really minor suggestions.

**Comment by nabobalis** [comment]
A short changelog as well and I think this is good to go.
Thanks for working on this @frederic-auchere

File: examples/advanced_wow.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+# First, we call WOW with now arguments, which returns the default WOW enhancement.
+
+wow_map = enhance.wow(input_map)
+
+###########################################################################
+# The we denoise the output using soft thresholding in the three first wavelet scales with sigma = 5, 2, 1
+
+denoise_coefficients = [5, 2, 1]
+wow_map_denoised = enhance.wow(input_map,
+                               denoise_coefficients=denoise_coefficients)
```
[the remark this answers is not in the document]
Comment (frederic-auchere) [reply examples/advanced_wow.py:46]: Wow (pun intended), I'm really sorry for the typos ...

**Comment by frederic-auchere** [comment]
Thanks Nabil !

File: examples/advanced_wow.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply examples/advanced_wow.py:46]: IT's all incredibly minor and when it comes to examples, it's really just personal preference (aka mine).

**Comment by nabobalis** [comment]
> Don't know anything about changelogs (but willing to learn !) Is there anything I should do ?

Sorry, I didn't see this edit. I pushed the changelog, so you can see how the system works.

**Comment by nabobalis** [comment]
The rendered version is here: https://sunpy--238.org.readthedocs.build/projects/sunkit-image/en/238/generated/gallery/advanced_wow.html#sphx-glr-generated-gallery-advanced-wow-py 

See if you like it @frederic-auchere and if so I think this is good to merge.

**Comment by nabobalis** [comment]
Thank you for the PR!

**Pull request merged**